src/contexts/currentUser.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, createContext, useState } from "react";
+import { useCurrentPatient } from "../fetchDataHooks";
+import { Skeleton } from "@mui/material";
+import api from "../capableApi";
+
+export const CurrentUserContext = createContext({
+  currentPatient: null,
+  currentActiveSubscription: null,
+});
+
+const findCurrentSubscription = (subscriptions) => {
+  return subscriptions.find((subscription) => {
+    return ["active", "trialing"].includes(subscription.status);
+  });
+};
+
+export const CurrentUserProvider = ({ children }) => {
+  const { currentPatient, isLoading: patientLoading } = useCurrentPatient();
+  const [currentActiveSubscription, setCurrentActiveSubscription] = useState(null);
+  const [subscriptionLoading, setSubscriptionLoading] = useState(true);
+  const [subscriptionError, setSubscriptionError] = useState(false);
+
+  useEffect(() => {
+    if (currentPatient?.id) {
+      (async () => {
+        try {
+          const response = await api.client.Subscription.list({
+            patientId: currentPatient.id,
+          });
+          if (response?.body) {
+            setCurrentActiveSubscription(findCurrentSubscription(response.body));
+          }
+          setSubscriptionLoading(false)
+        } catch (error) {
+          setSubscriptionError(error.message)
+          setSubscriptionLoading(false)
+        }
+      })();
+    }
+  }, [currentPatient?.id]);
+
+  if (patientLoading || subscriptionLoading) {
+    return <Skeleton variant="rectangular" height="100%" animation="wave" />;
+  }
+
+  if (subscriptionError) return <div>Woops something went wrong...</div>;
```

Comment:
Nitpick: I'm a strong backer of readability in code. I always find that wrapping my if statements in brackets is more readable
```suggestion
  if (subscriptionError) {
    return <div>Woops something went wrong...</div>;
  }
```